### PR TITLE
Use page objects for user-settings-form-test

### LINKED ISSUE #775: Rewrite integration tests for user-settings-form component to use page objects
Falls under umbrella issue #697.

## Resources

- What's a page object? [[1](http://martinfowler.com/bliki/PageObject.html),[2](https://wyeworks.com/blog/2015/5/13/using-the-page-object-pattern-with-ember-cli/)]
- [Quickstart guide](http://ember-cli-page-object.js.org/docs/v1.6.x/quickstart)
- See [`tests/pages/components/donations/create-donation.js`](https://github.com/code-corps/code-corps-ember/blob/develop/tests/pages/components/donations/create-donation.js) and [`tests/integration/components/donations/create-donation-test.js`](https://github.com/code-corps/code-corps-ember/blob/develop/tests/pages/components/donations/create-donation.js) for in-house examples of, respectively, a page object and the page object being utilized in an integration test.
- To generate a page object, run `ember generate page-object component/<component-name>` in the project directory.

### PR DESCRIPTION
**Pull request by sarupbanskota**
Closes #775 
Also under #697

### CODE REVIEW

File: tests/integration/components/user-settings-form-test.js [hunk 1]
```diff
@@ -2,46 +2,54 @@ import Ember from 'ember';
...
-  firstName: 'Test',
-  lastName: 'User',
-  twitter: '@testuser',
-  website: 'example.com',
-  biography: 'A test user'
+  firstName : 'Test',
+  lastName  : 'User',
+  twitter   : '@testuser',
+  website   : 'example.com',
+  biography : 'A test user'
```
Comment (JoshSmith) [inline tests/integration/components/user-settings-form-test.js]: I'm not sure if I'm a huge fan of this style of spacing here.

File: tests/integration/components/user-settings-form-test.js [hunk 2]
```diff
@@ -2,46 +2,54 @@ import Ember from 'ember';
...
 
 test('it renders', function(assert) {
   assert.expect(1);
 
-  this.render(hbs`{{user-settings-form}}`);
+  page.render(hbs`{{user-settings-form}}`);
 
-  assert.equal(this.$('.user-settings-form').length, 1);
+  assert.ok(page.userSettingsFormVisible);
 });
```
Comment (JoshSmith) [inline tests/integration/components/user-settings-form-test.js]: I'd actually remove the "it renders" test altogether.

File: tests/pages/components/user-settings-form.js [hunk 3]
```diff
@@ -10,13 +11,22 @@ import removeDoubleQuotes from '../../helpers/remove-double-quotes';
...
-  firstName: fillable('input[name=firstName]'),
-  lastName: fillable('input[name=lastName]'),
-  twitter: fillable('input[name=twitter]'),
-  website: fillable('input[name=website]'),
-  biography: fillable('input[name=biography]'),
+  firstName : fillable('input[name=firstName]'),
+  lastName  : fillable('input[name=lastName]'),
+  twitter   : fillable('input[name=twitter]'),
+  website   : fillable('input[name=website]'),
+  biography : fillable('input[name=biography]'),
```
Comment (JoshSmith) [inline tests/pages/components/user-settings-form.js]: Do we use these `fillable`'s anywhere right now, in this test?

**Comment by sarupbanskota** [comment]
@JoshSmith made relevant changes. 

about the key-spacing, are you opposed to it for a specific reason or because the rest of the app doesn't use it? if latter I was planning to open a PR to extend that to the entire repo.

**Comment by JoshSmith** [comment]
@sarupbanskota I think the keyspacing is additional overhead for people coming into a project. Visually it's nicer but think about every PR we'd ever have to help them understand the linting for. Think that the additional barrier to entry is probably not worth it. What do you think?

**Comment by JoshSmith** [comment]
Looks like your last commit broke tests. Would consider running tests before pushing up since feedback cycle is quicker that way.

**Pull request merged**